lib/layout/LayoutUtil.js

```diff
@@ -51,6 +52,36 @@ export function roundPoint(point) {
 }
 
 
+/**
+ * Convert the given connection bounds to a { top, left, bottom, right } descriptor.
+ *
+ * @param {Point|Connection} bounds
+ *
+ * @return {ConnectionTRBL}
+ */
+export function asConnectionTRBL(bounds) {
+  const waypoints = bounds.waypoints;
+  const points = waypoints.reduce((acc, point, index, array) => {
+    acc.x += point.x;
+    acc.y += point.y;
+
+    // Check if it's the last element
+    if (index === array.length - 1) {
+      acc.x /= array.length;
+      acc.y /= array.length;
+    }
+
+    return acc;
+  }, { x: 0, y: 0 });
+
+  return {
```

Comment:
Or would you like to compute the "mid" of an element?

Reply:
Yes, that's what we want to accomplish. I wasn't aware that you can calculate bounding box for a connection, I thought it's just for element.

Reply:
The issue is when you move the annotation from the flow that has label to it, and put on top of the label. The label should move downwards to the flow. 
![Oct-14-2024 19-12-12](https://github.com/user-attachments/assets/071f3a0b-e512-47f6-ab14-771d78d85119)



Currently, it's giving error cuz a connection is passed in asTRBL and returns NaN since it's not designed to handle connections. What my approach was to handle asTRBL to support connections as well.

Reply:
AdaptiveLabelPositioningBehavior in bpmn-js passes the flow element to get TRBL to facilitate positioning of label.